mtv-migration/mtv.sw.yaml

```diff
@@ -0,0 +1,80 @@
+specVersion: "0.8"
+id: mtv-migration
+version: '1.0'
+name: MTV migration workflow
+description: workflow executes a plan by creating a migration and waiting for it to be successful or failed
+annotations:
+  - "workflow-type/infrastructure"
+dataInputSchema: schemas/mtv-input.json
+start: CreateMigration
+functions:
+  - name: createMigration
+    type: rest
+    operation: 'specs/mtv.yaml#createMigration'
+  - name: getMigration
+    type: rest
+    operation: 'specs/mtv.yaml#getMigration'
+  - name: getMigrationError
+    type: expression
+    operation: .getStatusResult.status.conditions | map(select(.category == "Error" or .category == "Critical" or .type == "Failed"))[0].message
+states:
+  - name: CreateMigration
+    type: operation
+    actions:
+      - functionRef:
+          refName: createMigration
+          arguments:
+            namespace: .migrationNamespace
+            apiVersion: "forklift.konveyor.io/v1beta1"
+            kind: Migration
+            metadata:
+              name: .migrationName
+              namespace: .migrationNamespace
+            spec:
+              plan:
+                name: .migrationName
+                namespace: .migrationNamespace
+    transition: GetMigration
+  - name: GetMigration
+    type: operation
+    actions:
+      - name: getMigration
+        actionDataFilter:
+          toStateData: .getStatusResult
+        functionRef:
+          refName: getMigration
+          arguments:
+            namespace: .migrationNamespace
+            migration: .migrationName
+        sleep:
+          before: PT10S
+    transition: CheckMigrationStatus
+  - name: CheckMigrationStatus
+    type: switch
+    dataConditions:
+      - name: MigrationFailure
+        condition: .getStatusResult.status.conditions | map(select(.category == "Error" or .category == "Critical" or .type == "Failed")) | length > 0
+        transition: MigrationFailure
+      - name: MigrationSuccessful
+        condition: .getStatusResult.status.conditions | map(select(.type == "Succeeded")) | length > 0
```

Comment:
there is `|length > 0` for both conditions, so, the second one will never match if `length > 0` as the 1st condition will match first no?

Reply:
correct. Success conditions will not exist when there is a failure. also the other way around